projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionService.java

```diff
@@ -241,15 +245,15 @@ public void updateJobExecution(
 
       final Optional<com.vmware.taurus.service.model.DataJobExecution> dataJobExecutionPersistedOptional =
               jobExecutionRepository.findById(jobExecution.getExecutionId());
-      executionStatus = JobExecutionUtil.updateExecutionStatusBasedOnTerminationStatus(executionStatus, terminationMessage.getTerminationStatus());
 
       //This set contains all the statuses that should not be changed to something else if present in the DB.
       //Using a hash set, because it allows null elements, no NullPointer when contains method called with null.
       var finalStatusSet = new HashSet<>(List.of(ExecutionStatus.CANCELLED, ExecutionStatus.FAILED,
                                                  ExecutionStatus.FINISHED, ExecutionStatus.SKIPPED));
+      ExecutionStatus executionStatus = executionResult.getExecutionStatus();
 
       if (dataJobExecutionPersistedOptional.isPresent() &&
-              (dataJobExecutionPersistedOptional.get().getStatus() == executionStatus ||
+              (dataJobExecutionPersistedOptional.get().getStatus() == executionResult.getExecutionStatus() ||
                       finalStatusSet.contains(dataJobExecutionPersistedOptional.get().getStatus()))) {
          return;
```

Comment:
Let's log at debug to see how often it may happen

Reply:
Fixed